db/migrate/20160926083824_create_causes.rb

```diff
@@ -0,0 +1,10 @@
+class CreateCauses < ActiveRecord::Migration
+  def change
+    create_table :causes do |t|
+      t.string :shortcode
```

Comment:
Let's make both shortcode and name mandatory, and add validations in the model and here.